src/stactools/core/utils/raster_footprint.py

```diff
@@ -0,0 +1,284 @@
+import logging
+from itertools import groupby
+from typing import Any, Dict, Iterator, List, Optional, Tuple
+
+import numpy
+import numpy as np
+import rasterio
+import rasterio.features
+from pystac import Item
+from rasterio.crs import CRS
+from rasterio.warp import transform_geom
+from shapely.geometry import mapping, shape
+from shapely.geometry.multipolygon import MultiPolygon
+from shapely.geometry.polygon import Polygon
+
+logger = logging.getLogger(__name__)
+
+DEFAULT_PRECISION = 3
```

Comment:
As mentioned in a comment in the general conversation thread, this should probably be changed to 7 (centimeter) or 8 (millimeter).

Reply:
The geometry is always going to to be WGS84 geodetic coords, so I think we can just use a single value?

Reply:
Ah, I wasn't sure if the precision was used before or after reprojection, but if it's after, then I agree, will change to centimeter (7).